prover/src/constraints/evaluator/default.rs

```diff
@@ -233,8 +280,19 @@ where
             // evaluations buffer; we evaluate and compose constraints in the same function, we
             // can just add up the results of evaluating main and auxiliary constraints.
             evaluations[0] = self.evaluate_main_transition(&main_frame, step, &mut tm_evaluations);
-            evaluations[0] +=
-                self.evaluate_aux_transition(&main_frame, &aux_frame, step, &mut ta_evaluations);
+
+            // Make sure to only evaluate the aux transition if the user actually defined one
+            if !trace_aux_segment_has_only_lagrange_kernel_column(
+                self.air.context().lagrange_kernel_aux_column_idx(),
+                self.air.trace_info(),
+            ) {
+                evaluations[0] += self.evaluate_aux_transition(
+                    &main_frame,
+                    &aux_frame,
+                    step,
+                    &mut ta_evaluations,
+                );
+            }
```

Comment:
If there are no other aux constraints - shouldn't we just call `evaluate_fragment_main()` instead of calling `evaluate_fragment_full()` and then having this check?

Reply:
I since removed `trace_aux_segment_has_only_lagrange_kernel_column()`, since we don't want to support `Air` that only have a Lagrange kernel column (from previous comment). So the previous behavior where we call `evaluate_aux_transition()` unconditionally is now restored.